nixos/modules/services/network-filesystems/ipfs.nix

```diff
@@ -117,16 +120,15 @@ in
       after = [ "local-fs.target" ];
       before = [ "ipfs.service" "ipfs-offline.service" ];
 
+      environment.IPFS_PATH = cfg.dataDir;
+
       path  = [ pkgs.ipfs pkgs.su pkgs.bash ];
 
       preStart = ''
         install -m 0755 -o ${cfg.user} -g ${cfg.group} -d ${cfg.dataDir}
       '';
-
-      environment = pathEnv;
-
       script =  ''
-        if [[ ! -d ${cfg.dataDir}/.ipfs ]]; then
+        if [[ ! -d ${cfg.dataDir}/config ]]; then
```

Comment:
The standard IPFS implementation [does the same](https://github.com/ipfs/go-ipfs/blob/a6e96e6c99c4240427af6c3bd9a9ca1b4cea99b4/repo/fsrepo/fsrepo.go#L194-#L203) (for now)